chameleon-core/src/main/java/org/hibernate/omm/jdbc/MongoPreparedStatement.java

```diff
@@ -100,149 +106,112 @@ public void addBatch() {
 
     @Override
     public void setNull(final int parameterIndex, final int sqlType) {
-        parameters.put(parameterIndex, "null");
+        parameters.put(parameterIndex, BsonNull.VALUE);
     }
 
     @Override
     public void setBoolean(final int parameterIndex, final boolean x) {
-        parameters.put(parameterIndex, Boolean.toString(x));
+        parameters.put(parameterIndex, BsonBoolean.valueOf(x));
     }
 
     @Override
     public void setByte(final int parameterIndex, final byte x) {
-        parameters.put(parameterIndex, Byte.toString(x));
+        parameters.put(parameterIndex, new BsonInt32(x));
     }
 
     @Override
     public void setShort(final int parameterIndex, final short x) {
-        parameters.put(parameterIndex, Short.toString(x));
+        parameters.put(parameterIndex, new BsonInt32(x));
     }
 
     @Override
     public void setInt(final int parameterIndex, final int x) {
-        parameters.put(parameterIndex, "{ \"$numberInt\": \"" + x + "\" }");
+        parameters.put(parameterIndex, new BsonInt32(x));
     }
 
     @Override
     public void setLong(final int parameterIndex, final long x) {
-        parameters.put(parameterIndex, "{ \"$numberLong\": \"" + x + "\" }");
+        parameters.put(parameterIndex, new BsonInt64(x));
     }
 
     @Override
     public void setFloat(final int parameterIndex, final float x) {
-        parameters.put(parameterIndex, Float.toString(x));
+        parameters.put(parameterIndex, new BsonDouble(x));
     }
 
     @Override
     public void setDouble(final int parameterIndex, final double x) {
-        parameters.put(parameterIndex, "{ \"$numberDouble\": \"" + x + "\" }");
+        parameters.put(parameterIndex, new BsonDouble(x));
     }
 
     @Override
-    public void setBigDecimal(final int parameterIndex, @Nullable final BigDecimal x) {
-        parameters.put(parameterIndex, x == null ? "null" : "{ \"$numberDecimal\": \"" + x + "\" }");
+    public void setBigDecimal(final int parameterIndex, final BigDecimal x) {
```

Comment:
As far as I can tell, these parameters don't have to be nullable, as the intention is that callers will use `setNull` when they want to set a null value for a parameter.

Reply:
yeah, it seems so. i am a little bit worried about Hibernate's nullness robustness so I added it here. It is good move to treat it as non-null though.